components/raftstore/src/store/fsm/peer.rs

```diff
@@ -583,7 +595,6 @@ where
                             self.propose_batch_raft_command(true);
                         }
                     } else {
-                        self.propose_batch_raft_command(true);
```

Comment:
I don't think it's correct. For example, delete range request may be reordered to front of other puts/deletes as `propose_raft_command` may not propose any commands.